### PR TITLE
[Fix] `newline-after-import`: fix `exactCount` with `considerComments` false positive, when there is a leading comment

### PR DESCRIPTION
**Pull request by kinland**
This addresses #2882 and adds additional test cases to cover the combination of `exactCount` and `considerComments`

### CODE REVIEW

**Comment by kinland** [comment]
Sorry about the force-pushes... I'm a bit sleep deprived

**Comment by kinland** [comment]
I found an edge case that I missed... will push a new commit when I figure out how to fix it.

File: tests/src/rules/newline-after-import.js [hunk 1]
```diff
@@ -133,15 +133,30 @@ ruleTester.run('newline-after-import', require('rules/newline-after-import'), {
...
     },
     {
-      code: `import foo from 'foo';\n\n\nvar bar = 'bar';`,
+      code: `import foo from 'foo';\n\n// Some random comment\nvar bar = 'bar';`,
       parserOptions: { ecmaVersion: 2015, sourceType: 'module' },
       options: [{ count: 2, exactCount: true }],
     },
     {
-      code: `import foo from 'foo';\n\nvar bar = 'bar';`,
+      code: `import foo from 'foo';\n// Some random comment\nvar bar = 'bar';`,
```
Comment (ljharb) [inline tests/src/rules/newline-after-import.js:141]: instead of changing existing test cases, let's add new ones :-)

**Review by ljharb — CHANGES_REQUESTED** [review]

File: tests/src/rules/newline-after-import.js (hunk 1 above)
Comment (kinland) [reply tests/src/rules/newline-after-import.js]: as you wish! new commit just pushed

**Review by ljharb — APPROVED** [review]
LGTM assuming the new tests fail without the change, and pass with them. Thanks!

**Comment by reosarevok** [comment]
Thanks a lot! Clearly my one added test wasn't good enough when rebasing the previous PR, sorry.

**Comment by kinland** [comment]
> Thanks a lot! Clearly my one added test wasn't good enough when rebasing the previous PR, sorry.

Honestly, I can't fault you for that. I didn't consider that a comment *before* an import would break my code, either. These things get very complex quickly.
